mos_tests/neutron/python_tests/base.py

```diff
@@ -174,6 +174,26 @@ def check_ping_from_vm_helper(self, vm, vm_keypair, ip_to_ping,
                              vm_login=vm_login, vm_password=vm_password)
         return res
 
+    def check_ping_from_vm_with_ip(self, vm_ip, vm_keypair=None,
+                                   ip_to_ping=None, ping_count=1,
+                                   vm_login='cirros',
+                                   vm_password='cubswin:)'):
+        if ip_to_ping is None:
+            ip_to_ping = [settings.PUBLIC_TEST_IP]
+        if isinstance(ip_to_ping, six.string_types):
+            ip_to_ping = [ip_to_ping]
+        pkeys = self.convert_private_key_for_vm([vm_keypair.private_key])
+        cmd_list = ["ping -c{0} {1}".format(ping_count, x) for x in ip_to_ping]
+        with self.env.get_ssh_to_vm(vm_ip, private_keys=pkeys,
+                                    username=vm_login,
+                                    password=vm_password) as vm_remote:
+            res = vm_remote.execute(' && '.join(cmd_list))
```

Comment:
Yes, in this case it is better to use execute and assert in order to fail test and not to throw exception.